### DISCUSSION TITLE
Does Saleor Dashboard Support Bulk Listing of Products

### DISCUSSION THREAD

**Discussion by kk2211** [mirumee/saleor-dashboard#2203] [Q&A] [answered]
Hi , Is there a way of uploading products in bulk via a CSV(or an api call)?
We have a 1000+ products, uploading them manually one by one will be a hassle.

**Comment by timuric** [comment]
@kk2211 we recommend uploading products via API calls. Everything you see in the dashboard works via API, so you can automate it.
To explore how dashboard is creating a product you can simply use devtools inspector and look at network requests, or use a dedicated graphql inspector
https://chrome.google.com/webstore/detail/graphql-network-inspector/ndlbedplllcgconngcnfmkadhokfaaln?hl=en-GB

**Comment by MahmadSharaf** [comment]
I can suggest this project, https://github.com/grll/saleor-gql-loader. It is a bit outdated, but you can work yourself out with it. I have been using it for bulk listing

**Comment by timuric** [comment, marked as answer]
Please take a look at this mutation https://docs.saleor.io/docs/3.x/api-reference/mutations/product-variant-bulk-create
